File: src/lib/tauri/credentials.ts
import { invoke, isTauri } from '@tauri-apps/api/core';

// Browser-Fallback nur für die Entwicklung ohne Tauri; in der App landet nichts hier.
const devStore = new Map<string, string>();

export async function getCredential(key: string): Promise<string | null> {
  if (!isTauri()) return devStore.get(key) ?? null;
  return invoke<string | null>('credentials_get', { key });
}

export async function setCredential(key: string, secret: string): Promise<void> {
  if (!isTauri()) {
    devStore.set(key, secret);
    return;
  }
  await invoke('credentials_set', { key, secret });
}

export async function deleteCredential(key: string): Promise<void> {
  if (!isTauri()) {
    devStore.delete(key);
    return;
  }
  await invoke('credentials_delete', { key });
}

File: src-tauri/src/commands/credentials.rs
#[tauri::command]
pub fn credentials_get(key: String) -> Result<Option<String>, String> {
    crate::credentials::get(&key)
}

#[tauri::command]
pub fn credentials_set(key: String, secret: String) -> Result<(), String> {
    crate::credentials::set(&key, &secret)
}

#[tauri::command]
pub fn credentials_delete(key: String) -> Result<(), String> {
    crate::credentials::delete(&key)
}
